src/Components/GrowthChart/GrowthChartOptions/GrowthChartAnnotations.tsx

```diff
@@ -0,0 +1,47 @@
+import i18n from '@dhis2/d2-i18n';
+
+export interface AnnotationLabel {
+    display: boolean;
+    content?: () => string;
+    position?: 'top' | 'bottom' | 'center' | 'start' | 'end';
+    yAdjust?: number;
+}
+
+interface Annotation {
+    display: boolean;
+    type: 'line';
+    scaleID?: string;
+    value?: number;
+    borderWidth?: number;
+    label?: AnnotationLabel;
+}
+
+export const GrowthChartAnnotations = (xAxisValues: number[]) => {
+    const annotations: Annotation[] = [];
+
+    xAxisValues.forEach(
+        (label, index) => {
+            if (label % 12 === 0 && label !== 0) {
+                annotations.push({
+                    display: true,
+                    type: 'line',
+                    scaleID: 'x',
+                    borderWidth: 2,
+                    value: index,
+                    label: {
+                        display: true,
+                        content: () => {
+                            const value = label / 12;
+                            if (value === 1) return `${value} ${i18n.t('Year')}`;
+                            return `${value} ${i18n.t('Years')}`;
+                        },
+                        position: 'end',
+                        yAdjust: 10,
+                    },
+
+                });
+            }
+        },
+    );
+    return annotations;
```

Comment:
The `map function` automatically returns a new array, so there is no need to declare an empty array first and then use push to fill the array. Therefore, I have a suggestion:
```suggestion
    const annotations: Annotation[] = xAxisValues
        .filter((label) => label % 12 === 0 && label !== 0)
        .map((label) => ({
            display: true,
            type: 'line',
            scaleID: 'x',
            borderWidth: 2,
            value: label,
            label: {
                display: true,
                content: () => {
                    const value = label / 12;
                    if (value === 1) return `${value} ${i18n.t('Year')}`;
                    return `${value} ${i18n.t('Years')}`;
                },
                position: 'end',
                yAdjust: 10,
            },
        }));
    return annotations;
```